python/tvm/relay/prelude.py

```diff
@@ -89,26 +89,32 @@ def _get_name_static(canonical, dtype, shape):
         shape_str = "scalar"
     if canonical == "tensor_t":
         return "static_tensor_{}_{}_t".format(dtype, shape_str)
-    return "{}_{}_{}".format(canonical, dtype, shape_str)
+    if not batch_dim or canonical == "tensor_constructor" or canonical == "tensor_nil":
```

Comment:
Be careful that `bool(batch_dim=0)` also gives you false. Better to use `batch_dim is None` if that's the desire semantic.

```suggestion
    if batch_dim is None or canonical in ["tensor_constructor", "tensor_nil"]:
```